### PR TITLE
[core] support for molecular diamag contribution

### PR DESCRIPTION
**Pull request by kpwydra**
## Goals

 - Create a list of SMILES strings for the molecules reported in reference 10.1021/ed085p532 (Tables 3–5).
 - Develop an RDKit-based query tool that checks whether the current SDF file contains the molecules from the SMILES list.
 - If a given molecule is found in the SDF file, the diamagnetic contribution for that species will be included, while addition of atomic Pascal constants will be excluded for given molecule.

ISSUE: https://github.com/mag-bros/mag-bridge/issues/64

### CODE REVIEW

File: data/sdf/S2O32-.sdf
Comment (Sevelantis) [inline data/sdf/S2O32-.sdf]: test files should be in tests directory, even if they fail - we need to address this. Could you move all test files to tests dir

File: src/constants.py [hunk 1]
```diff
@@ -29,6 +30,444 @@ def GetRelevantRingAtoms() -> list[str]:
...
+@note4: RDKit does not support SMILES with double bonds for hypervalent halogen oxoanions. 
+As a result, the generated SMILES for BrO3-, ClO3-, ClO4-, IO3- and IO4- anions do not reflect their correct structures.
+@note5: For BF4- anion, the negative charge must be located on the boron atom. Otherwise SDF fails to parse. 
+@note6: In the reference (Table 3), PO4(3-) anion was incorrectly written as PO3(3-).
+@note7: PtCl6(2-) (Table 3 in the reference) was intentionally omited in the dictionary.
+@note8: Our software does not support stereochemical structures.
+@note9: carbonyl (CO) ligand must be drawn with (+) charge located on C and (-) on O. Otherwise, SDF fails to parse.
+
+Here's a breakdown of the values:
+"""
```
Comment (Sevelantis) [inline src/constants.py]: its a very long read, consider shrinking to 4-5 lines

File: src/constants.py [hunk 2]
```diff
@@ -29,6 +30,444 @@ def GetRelevantRingAtoms() -> list[str]:
...
+        "SMILES": "O=S(=O)([O-])[O-]",
+        "DIAMAG_SUS": -40.1,
+    },
+    "S2O3(2-)": {
+        "Name": "thiosulfate",
+        "SMILES": {  # Two canonical SMILES of this anion exist.
+            "O=S([O-])([O-])=S",
+            "O=S(=O)([O-])[S-]",
+        },
+        "DIAMAG_SUS": -46,
```
Comment (Sevelantis) [inline src/constants.py]: I recommend aligning an overall better structure to handling corner cases like this - it is not consistent if other entries are of type "str" and this one is of type "set" - make them either all string, or all set, or choose a different solution

File: tests/unit/test_molecule_match.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        # Test Results
+        try:
+            # trunk-ignore(bandit/B101)
+
+            # Exception (thiosulfate anion)
+            if compound.source_file in EXCEPTION:
+                print(
+                    f'[INF] "{compound.source_file}": ⚠ Allowed {len(set(group))} canonical SMILES case: {set(group)}'
+                )
+                continue
```
Comment (Sevelantis) [inline tests/unit/test_molecule_match.py]: It's good to know we have corner cases, but this whole if statement and EXCEPTION variable should be removed in the final version.

File: data/sdf/S2O32-.sdf
Comment (kpwydra) [reply data/sdf/S2O32-.sdf]: The S2O32-.sdf file in the data/sdf folder is an accidental copy and should be removed. I have already deleted it locally.

Comment (Sevelantis) [reply data/sdf/S2O32-.sdf]: What I'm trying to say is we need to better organize our SDF files, e.g. per use-case. From what you say it looks like the whole `data/sdf` directory will be used for tests only. We need better SDF organization, e.g. can you distinguish logical groups? Maybe per-feature?
